IMBLightroom7Parser.m

```diff
@@ -638,7 +638,7 @@ + (IMBLightroomNodeType)nodeTypeForCreationId:(NSString *)creationId
 - (FMDatabasePool*) createLibraryDatabasePool
 {
 	NSString* databasePath = [self.mediaSource path];
-	FMDatabasePool* databasePool = [[FMDatabasePool alloc] initWithPath:databasePath flags:SQLITE_OPEN_READONLY vfs:@"unix-none"];
+	FMDatabasePool* databasePool = [[FMDatabasePool alloc] initWithPath:databasePath flags:SQLITE_OPEN_READWRITE vfs:@"unix-none"];
```

Comment:
I'm afraid, this may lead to a corrupt Lightroom catalog. Consider the following scenario:

1. Open iMedia Browser and show content of Lightroom library (will create the -shm and -wal temporary files (-wal file size == 0 Byte))
2. Open Lightroom (will remove -shm file and replace -wal file (-wal file size == 4KB))
3. Exec any query on Lightroom catalog within iMedia Browser (oops, will shrink -wal file to 0 Byte)
4. Delete an image within Lightroom (remove reference)
5. Quit Lightroom (will remove -wal file)
6. Exec any query on Lightroom catalog within iMedia Browser: will result in error „file is encrypted or is not a database“
7. Try reopen Lightroom: will give an error like „Could not open catalog because of unexpected error“
9. Lightroom catalog is corrupt ⚡️

From all what I have analyzed so far, it is quite clear that the Lightroom 7 (Classic CC) catalog is setup with  `journal_mode=WAL` and `locking_mode=EXCLUSIVE`. This has the potential of corrupting the catalog if we open it READWRITE ourselves as described above. Thus, I don't see any other option than to stick to READONLY while accepting the drawback that due to WAL we will only have access to the catalog while Lightroom is running (i.e. keeps a connection to the catalog). Consequently, this affordance must be clearly communicated to the iMedia user (probably via clicking on the warning label).

Reply:
Bummer.

The source of the problem is that the database is setup to use a WAL journal. When we attempt to open it in READ_ONLY mode, SQLite still needs to create the WAL journal and fails because it lacks write right. 

According to SQLite documentation, WAL databases cannot be opened in READ_ONLY mode. It appears that this nonetheless worked and allowed us to read the catalog while Lightroom is running and has created the WAL file. It however seems likely that we will run into trouble with this approach.

The best approach may be to create a read-only copy of the database. Much like we did for Lightroom3. Only this time we need to also copy the shm and wal files. If there are concurrent changes in Lightroom, there however is a risk that we will end up with an incosnistent combination of the 3 files. Still seems to be the safest approach.